File: Java/src/onreadyfyMe/Moto.java
package onreadyfyMe;

import java.text.DecimalFormat;

public class Moto extends Vehiculo {
	private String cilindrada;

	public String getCilindrada() {
		return cilindrada;
	}

	public void setCilindrada(String cilindrada) {
		this.cilindrada = cilindrada;
	}

	public Moto(String marca, String modelo, double precio, String cilindrada) {
		super(marca, modelo, precio);
		this.cilindrada = cilindrada;
	}

	public Moto() {
	}

	@Override
	public void mostrarTodosDatos() {
		DecimalFormat formatea = new DecimalFormat("###,###.00");
		System.out.println("Marca: " + getMarca() + " // Modelo: " + getModelo() + " // Cilindrada: " + getCilindrada()
				+ " // Precio: $" + formatea.format(getPrecio()));
	}

}

File: Python/moto.py
from vehiculo import Vehiculo

class Moto(Vehiculo):
    __cilindrada = 0

    def __init__(self, marca, modelo, precio, cilindrada):
        Vehiculo.__init__(self, marca, modelo, precio)
        self.__cilindrada = cilindrada

    def get_cilindrada(self):
        return self.__cilindrada

    def set_cilindrada(self, cilindrada):
        self.__cilindrada = cilindrada

    def __str__(self):
        return f"Marca: {self.get_marca()} // Modelo: {self.get_modelo()} // Puertas: {self.get_cilindrada()} // Precio: ${'{:,.2f}'.format(self.get_precio()).replace(',','~').replace('.',',').replace('~','.')}"

    def __gt__(self, moto):
        return (self.get_precio() > moto.get_precio())
    
    def esLujo(self):
        if int(self.__cilindrada[:3]) >= 150:
            return True
        else:
            return False